packages/api/login.go

```diff
@@ -145,7 +145,15 @@ func login(w http.ResponseWriter, r *http.Request, data *apiData, logger *log.En
 	if err != nil {
 		return errorAPI(w, err, http.StatusInternalServerError)
 	}
-	notificator.AddUser(wallet, state)
+
+	recipient := &notificator.Recipient{ID: wallet, EcosystemID: state}
+	notificator.AddRecipient(recipient)
+
+	// if exists VDE, then add the recipient in VDE
+	if result.IsVDE {
+		recipient.IsVDE = true
+		notificator.AddRecipient(recipient)
```

Comment:
Are you sure that it's correct? You have a recipient pointer, add it once to notificator, modify IsVde field, then add second time the same pointer to notificator. If the correct behaviour to receive different notifications to vde and non vde recipients, you should use a clone of that struct.

Reply:
It was not obvious, but it worked only for passing parameters to the function. Fixed using without a pointer in this commit e839acc